src/main/java/org/zalando/nakadi/service/EventStream.java

```diff
@@ -122,6 +123,23 @@ public void streamEvents(final AtomicBoolean connectionReady, final Runnable che
                         batchStartTimes.put(partition, currentTimeMillis());
                     }
                 }
+                // Dump some data that is exceeding memory limits
+                long memoryUsed = currentBatches.values().stream()
+                        .mapToLong(partition -> partition.stream().mapToLong(event -> event.length).sum()).sum();
```

Comment:
This will actually be run at least 10 times a second. Do you think it will make sense to somehow limit the period of running this calculation?

Reply:
Optimized